Wrappers/Python/ccpi/optimisation/operators/GradientOperator.py

```diff
@@ -39,29 +60,80 @@ class Gradient(LinearOperator):
                 Example (2D): u\in X, \nabla(u) = [\partial_{y} u, \partial_{x} u]
                               u^{*}\in Y, \nabla^{*}(u^{*}) = \partial_{y} v1 + \partial_{x} v2
         
-                Grad_order = ['channels', 'direction_z', 'direction_y', 'direction_x']
-                Grad_order = ['channels', 'direction_y', 'direction_x']
-                Grad_order = ['direction_z', 'direction_y', 'direction_x']
-                Grad_order = ['channels', 'direction_z', 'direction_y', 'direction_x']
-                
-
     '''
+
+    #kept here for backwards compatability
+    CORRELATION_SPACE = CORRELATION_SPACE
+    CORRELATION_SPACECHANNEL = CORRELATION_SPACECHANNEL
+
+    def __init__(self, gm_domain, bnd_cond = 'Neumann', **kwargs):
+        """Constructor method
+        """        
+        super(Gradient, self).__init__() 
+
+        backend = kwargs.get('backend',C)
+
+        correlation = kwargs.get('correlation',CORRELATION_SPACE)
+
+        if correlation == CORRELATION_SPACE and gm_domain.channels > 1:
+            #numpy implementation only for now
+            backend = NUMPY
+            "Warning: correlation='Space' on multi-channel dataset will use `numpy` backend"
```

Comment:
this should be done with[ `warnings`](https://docs.python.org/3/library/warnings.html#warnings.warn)
```python

import warnings

warnings.warn("Warning: correlation='Space' on multi-channel dataset will use `numpy` backend")
```